src/hooks/use-pagination.ts

```diff
@@ -42,6 +42,16 @@ const usePagination: UsePagination = (layoutService) => {
     }));
   }, [layoutService.layout.qHyperCube.qSize.qcy, setPageInfo]);
 
+  useEffect(() => {
+    const { currentPage, totalPages } = pageInfo;
+
+    // currPage is base 0 and totalPages always includes remainder rows in last page
+    // so we need to consider both of them for prevent landing in missing page
+    if (currentPage + 1 > totalPages) {
+      setPageInfo({ ...pageInfo, currentPage: totalPages - 1 });
+    }
```

Comment:
You could also go back to the first page. But I think this is better. Would it make sense to do this for selections as well? Would be nice to simplify and have one solution. We are breaking the pattern that sn-table follows though. Whenever the number of rows change, we go back to the first page. What do you think?

Reply:
whenever number of rows change, "in case of selection" it definitely makes sense to go to the first page (which both table and PV is respecting, after the other PR is being merged)

but what makes PT different than table is **the expansion of a row**, because if that is the case (row changes -> always go to the first page) expanding a row (in page 3 for example) would land user in first page, which we dont want that to happen.

This check is only addressing the collapse scenario where user can still remains in a page that is no more existing.

Reply:
Agree, I think this is the most natural way to do it